homework/index.js

```diff
@@ -30,15 +32,95 @@
     return elem;
   }
 
+  function clearContainer(container) {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function renderError(err) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  }
+
+  function addRow(tbody, label, value = '') {
+    const row = createAndAppend('tr', tbody);
+    createAndAppend('td', row, { text: `${label} :`, class: 'label' });
+    createAndAppend('td', row, { text: value });
+    return row;
+  }
+
+  function createInfoList(repository, container) {
+    clearContainer(container);
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
+    });
+    addRow(tbody, 'Description:', repository.description);
+    addRow(tbody, 'Forks:', repository.forks);
+    addRow(tbody, 'Updated:', new Date(repository.updated_at).toLocaleString());
+  }
+
+  function createContributorsList(repository, ul) {
+    clearContainer(ul);
+    fetchJSON(repository.contributors_url)
+      .then(contributors =>
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul);
+          const a = createAndAppend('a', li, {
+            href: repository.html_url,
+            target: '_blank',
+            id: 'link_contributions',
+          });
+          const div = createAndAppend('div', a);
+          createAndAppend('img', div, {
+            class: 'contributor_avatar',
+            src: contributor.avatar_url,
+            alt: contributor.login,
+          });
+          createAndAppend('p', div, { id: 'p1', text: contributor.login });
+          createAndAppend('p', div, { id: 'p2', text: contributor.contributions });
```

Comment:
As I mentioned in the week1 feedback, you should not use `id` attributes here. Because you are creating these `p` elements in a (`.forEach()`) loop, these same `id` attributes occur more than one time. And `id` attributes must be unique. Therefore the HTML you generate is invalid. You should use CSS classes instead. They can be reused many times in the HTML page.

```html
<ul id="Contributors_list">
  <li>
    <a href="https://github.com/HackYourFuture/alumni" target="_blank" id="link_contributions"
      ><div>
        <img
          class="contributor_avatar"
          src="https://avatars0.githubusercontent.com/u/15912395?v=4"
          alt="gijscor"
        />
        <p id="p1">gijscor</p>
        <p id="p2">47</p>
      </div></a
    >
  </li>
  <li>
    <a href="https://github.com/HackYourFuture/alumni" target="_blank" id="link_contributions"
      ><div>
        <img
          class="contributor_avatar"
          src="https://avatars1.githubusercontent.com/u/37488847?v=4"
          alt="wouterkleijn"
        />
        <p id="p1">wouterkleijn</p>
        <p id="p2">11</p>
      </div></a
    >
  </li>
</ul>
```